Rubberduck.Parsing/Symbols/ModuleBodyElementDeclaration.cs

```diff
@@ -117,10 +73,78 @@ protected void AddParameters(IEnumerable<ParameterDeclaration> parameters)
             _parameters.AddRange(parameters);
         }
 
+        private Lazy<(bool IsInterfaceImplementation, ClassModuleDeclaration InterfaceImplemented)> _interface;
+
+        /// <summary>
+        /// Returns true if this member is a concrete implementation of an interface.
+        /// </summary>
+        public bool IsInterfaceImplementation => _interface != null && _interface.Value.IsInterfaceImplementation;
+
+        /// <summary>
+        /// Returns the interface that this member implements from, or null if not an implementation.
+        /// </summary>
+        public ClassModuleDeclaration InterfaceImplemented => _interface?.Value.InterfaceImplemented;
+
+        private Lazy<Declaration> _implemented;
+
+        /// <summary>
+        /// Returns the interface member that this member is a concrete implementation of, or null if not an implementation.
+        /// </summary>
+        public Declaration InterfaceMemberImplemented => _implemented?.Value;
+
+        /// <inheritdoc/>
+        public string ImplementingIdentifierName => this.ImplementingIdentifierName();
+
+        /// <summary>
+        /// Returns true if this Declaration is a concrete implementation of the passed member.
+        /// </summary>
+        /// <param name="interfaceMember">The member to test for implementation of.</param>
+        /// <returns>True if this Declaration is a concrete implementation of interfaceMember.</returns>
+        protected abstract bool Implements(ICanBeInterfaceMember interfaceMember);
```

Comment:
The XML-Doc here makes me wonder whether `ICanBeInterfaceMember` is a misnomer.

Just because an InterfaceMember has not been implemented that doesn't mean it's not an interface member anymore... This distinction only makes sense in the context of the implementing class module.

Could you clarify which context `ICanBeInterfaceMember` should be used in?

Reply:
I borrowed the naming convention for `ICanBeInterfaceMember` from `ICanBeDefaultMember`, but I can't claim to be thrilled with either of them.  It's basically anything that can be exposed on a VB interface. The `ICanBeInterfaceMember` exists because public variables in interface classes are members, so `VariableDeclaration` and `ModuleBodyElementDeclaration` need the functionality that it provided.  I'll put XML-Doc on the interface itself, but any naming suggestions would be welcome.

Reply:
how about `IInterfaceExposable`?

Reply:
I like that. Done.